cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/ApiCredentials.java

```diff
@@ -0,0 +1,246 @@
+import io.strimzi.operator.cluster.model.ModelUtils;
+import io.strimzi.operator.common.InvalidConfigurationException;
+import io.strimzi.operator.common.Util;
+import io.strimzi.operator.common.model.InvalidResourceException;
+import io.strimzi.operator.common.model.Labels;
+import io.strimzi.operator.common.model.PasswordGenerator;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.AUTH_FILE_KEY;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.HEALTHCHECK_PASSWORD_KEY;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.HEALTHCHECK_USERNAME;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.REBALANCE_OPERATOR_PASSWORD_KEY;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.REBALANCE_OPERATOR_USERNAME;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.TOPIC_OPERATOR_PASSWORD_KEY;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.TOPIC_OPERATOR_USERNAME;
+import static io.strimzi.operator.common.model.cruisecontrol.CruiseControlApiProperties.TOPIC_OPERATOR_USERNAME_KEY;
+
+/**
+ * Uses information in a Kafka Custom Resource to generate a API credentials configuration file to be used for
+ * authenticating to Cruise Control's REST API.
+ */
+public class ApiCredentials {
+    private static final List<String> FORBIDDEN_USERNAMES = Arrays.asList(
+            HEALTHCHECK_USERNAME,
+            REBALANCE_OPERATOR_USERNAME,
+            TOPIC_OPERATOR_USERNAME
+    );
+
+    private final ApiUsers apiUsers;
+    private final String userManagedApiSecretName;
+    private final String userManagedApiSecretKey;
+    private final String namespace;
+    private final String cluster;
+    private final Labels labels;
+    private final CruiseControlSpec ccSpec;
+    private final OwnerReference ownerReference;
+
+    /**
+     * Constructs the Api Credentials Model for managing API users for Cruise Control API.
+     *
+     * @param namespace         Namespace of the cluster
+     * @param cluster           Name of the cluster to which this component belongs
+     * @param labels            Labels for Cruise Control instance
+     * @param ownerReference    Owner reference for Cruise Control instance
+     * @param ccSpec         Custom resource section configuring Cruise Control API users
+     */
+    public ApiCredentials(String namespace, String cluster, Labels labels, OwnerReference ownerReference, CruiseControlSpec ccSpec) {
+        this.namespace = namespace;
+        this.cluster = cluster;
+        this.labels = labels;
+        this.ownerReference = ownerReference;
+        this.ccSpec = ccSpec;
+        this.apiUsers = ApiUsers.create(ccSpec);
```

Comment:
Are we using this field? It looks like that wherever you need to get the `ApiUsers` instance you access the CC spec with `getApiUsers()` which is the logical way, or? So do we really need the `ApiUsers.create()` factory method?

Reply:
Currently fixing this, I'll note here once it's ready

Reply:
Refactored, let me know how it looks to you